src/Setup.ts

```diff
@@ -0,0 +1,46 @@
+// This file is run before main.js to setup the tools that the action depends on
+// https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runspre
+
+import * as core from '@actions/core';
+import * as tc from '@actions/tool-cache';
+
+export const sqlcmdToolName = 'go-sqlcmd';
+export const sqlcmdVersion = '0.8.1';
+
+export default class Setup {
+    
+    /**
+     * Ensures go-sqlcmd is in the runner's tool cache and PATH environment variable.
+     */
+    public static async setupSqlcmd(): Promise<void> {
+        // Get sqlcmd from tool cache; if not found, download it and add to tool cache
+        let sqlcmdPath = tc.find(sqlcmdToolName, sqlcmdVersion);
+        if (!sqlcmdPath) {
+            const extractedPath = await this.downloadAndExtractSqlcmd();
+            sqlcmdPath = await tc.cacheDir(extractedPath, sqlcmdToolName, sqlcmdVersion);
+        }
+        
+        // Add sqlcmd to PATH
+        core.addPath(sqlcmdPath);
```

Comment:
does sqlCmd get added to PATH again if it's already there from before?

Reply:
According to the [docs](https://github.com/actions/toolkit/tree/main/packages/core#path-manipulation), the change should only be for the duration of the job.

>To make a tool's path available in the path for the remainder of the job (without altering the machine or containers state), use addPath